### ISSUE TITLE
Installing with database connection via unix socket?

### ISSUE THREAD

**Issue by AsavarTzeth** [ampache/ampache#790]
I have searched the web and the official docs (wiki?), here. However, I cannot find anything close to a good description of how I can configure this in either the web gui installation tool, or manually in the config file. It does not help that the search engines seem to think I want information about apache.

My database is already setup & configured correctly. I have checked, all is set. The only thing I am missing is information on how I configure ampache.

My system:
Ubuntu Server 14.10
MariaDB 5.5.41
Nginx 1.6.2 (Using your template, with missing content filled in)
PHP (FPM) 5.5.12
Ampache 7.3 (latest)

Database is listening on `/var/run/mysqld/mysqld.sock` and using `auth_socket` plugin for authentication with www-data system user.

Could anyone help me out? I would be happy to contribute in some way to the project, if I can just get this up and running first.

**Comment by Afterster** [comment]
I really don't know about that, never used auth_socket.
According to line 250 in /lib/class/dba.class.php it looks like it could be supported if you prefix the hostname with a `/`, but I'm afraid you will have to test it.

**Comment by AsavarTzeth** [comment]
I did test it, but I did not manage to get it working, at least not during setup. I have now got it working the traditional way and I could make a second attempt right now. What worries me is this might prevent it from working; extract from ampache.cfg.php:
```
; Password for your ampache database, this can not be blank
; this is a 'forced' security precaution, the default value
; will not work
; DEFAULT: ""
database_password = "XxXxXxXxX"
```
I would have to do the test with a password filled in, because of this "security precaution". I find this funny since I am of the opinion that simply trusting the web server user directly should be more secure than storing the password in a file, which requires greater trust in the application itself (I could be wrong though).

I will do some more testing and share my results.

**Comment by AsavarTzeth** [comment]
Okay initial testing is looking good. The ampache debug log provided me with some insight.
```
2015-04-03 15:36:26 [AMPACHEUSER] (Dba) -> Error: ["42S02",1146,"Table 'DATABASE.localplay_mpd' doesn't exist"] 
2015-04-03 15:36:26 [AMPACHEUSER] (Dba) -> Error: ["42S02",1146,"Table 'DATABASE.localplay_mpd' doesn't exist"] 
2015-04-03 15:36:26 [AMPACHEUSER] (Dba) -> Error: ["42S02",1146,"Table 'DATABASE.localplay_mpd' doesn't exist"] 
2015-04-03 15:36:26 [AMPACHEUSER] (Dba) -> Error: ["42S02",1146,"Table 'DATABASE.localplay_mpd' doesn't exist"]
```
Now I checked the tables, this is true *.localplay_mpd does not exist. But that is okay. My reasoning here though is there would be a whole lot more Dba errors if my switch had failed.

The only thing left for me now is figuring out a more sure way of knowing ampache did not fall back on some old settings stored somewhere. In other words I really need MariaDB to tell me which user did in fact connect now. Too bad the general logging feature never seems to work whenever I enable it.

**Comment by Afterster** [comment]
Mhh ok thanks for  your feedback. I did a short update to allow socket authentication on web setup, if you want to try again the web setup.

**Comment by AsavarTzeth** [comment]
Sure, anyway I can be of help. I will see if I can find the time to do a test tomorrow. If this helps future deployments it will sure be worth it.

Other than this I must say I love the software so far. I have only had a few tiny issues, but nothing too serious. Only one thing that might be a bug. I will open a new issue for that.

**Comment by AsavarTzeth** [comment]
Okay, I have now installed and tested on a fresh vm. I duplicated as much as possible from my initial setup and installed ampache with git clone and checked out commit 9430729e3.

Now the during the Ampache Installation it accepts my configuration when I click write. However, __"config/ampache.cfg.php configured?"__ still says __ERROR__. I assume this means there is another check somewhere that thinks the config file is incorrect?
